packages/flutter_tools/lib/src/test/flutter_platform.dart

```diff
@@ -25,8 +25,10 @@ const String _kPath = '/runner';
 
 String shellPath;
 
-void installHook() {
-  hack.registerPlatformPlugin(<TestPlatform>[TestPlatform.vm], () => new FlutterPlatform());
+void installHook({ int observatoryPort }) {
+  hack.registerPlatformPlugin(
+      <TestPlatform>[TestPlatform.vm],
+      () => new FlutterPlatform(observatoryPort: observatoryPort));
 }
```

Comment:
CoverageCollector assumes we only ever set it when coverage is enabled, but looks like that's not the case, so scratch that. Will go with a property on FlutterPlatform and just re-implement the one-liner.

Also need to have it pick a random port per call to startTest to preserve the existing behaviour and make it play nice with current tests.